### PR TITLE
Porting variational

### PR DESCRIPTION
**Pull request by GreatRSingh**
## Description

Fix #(issue)

Ports Variational Randomiser.
Tasks:
- [x] Port Class.
- [x] Write Unit Tests. (Equivalence Checking Unit Tests.)
- [x] Add to Docs.
- [x] Add to init

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/layers.py [hunk 1]
```diff
@@ -4794,6 +4798,46 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
         return output
 
 
+class VariationalRandomizer(nn.Module):
+    """Add random noise to the embedding and include a corresponding loss."""
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:4802]: Can you add a usage example and more description? Aim for about a paragraph

File: deepchem/models/torch_models/layers.py [hunk 2]
```diff
@@ -4794,6 +4798,46 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
...
+
+    def __init__(self, embedding_dimension: int, annealing_start_step: int,
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:4804]: Needs docstring

File: deepchem/models/torch_models/layers.py [hunk 3]
```diff
@@ -4794,6 +4798,46 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
...
+        super(VariationalRandomizer, self).__init__(**kwargs)
+        self._embedding_dimension = embedding_dimension
+        self._annealing_final_step = annealing_final_step
+        self._annealing_start_step = annealing_start_step
+        self.dense_mean = nn.LazyLinear(embedding_dimension)
+        self.dense_stddev = nn.LazyLinear(embedding_dimension)
+        self.combine = CombineMeanStd(training_only=True)
+        self.loss_list: List = list()
+
+    def forward(self, inputs, training=True):
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:4815]: Type annotations/doc strings

File: deepchem/models/torch_models/layers.py [hunk 4]
```diff
@@ -4794,6 +4798,46 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
         return output
 
 
+class VariationalRandomizer(nn.Module):
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:4801]: Make sure to add to layers.rst and torch_layers.csv

Comment (rbharath) [inline deepchem/models/torch_models/layers.py:4801]: Needs unit tests

Comment (GreatRSingh) [reply deepchem/models/torch_models/layers.py:4801]: Done

Comment (GreatRSingh) [reply deepchem/models/torch_models/layers.py:4801]: Done

File: deepchem/models/torch_models/layers.py (hunk 2 above)
Comment (GreatRSingh) [reply deepchem/models/torch_models/layers.py:4830]: Done

File: deepchem/models/torch_models/layers.py (hunk 1 above)
Comment (GreatRSingh) [reply deepchem/models/torch_models/layers.py]: Done

File: deepchem/models/torch_models/layers.py (hunk 3 above)
Comment (GreatRSingh) [reply deepchem/models/torch_models/layers.py]: Done

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**